services/apps/alcs/src/portal/notice-of-intent-submission/notice-of-intent-submission.service.ts

```diff
@@ -61,6 +61,21 @@ export class NoticeOfIntentSubmissionService {
     @InjectMapper() private mapper: Mapper,
   ) {}
 
+  async getOrFailByFileNumber(fileNumber: string) {
```

Comment:
You should be okay to just use .findOneOrFail and remove your manual checking. You also need to add a test for this function.

Reply:
Updated!